File: c#/builder/builder/HtmlFormBuilder.cs
using System;
using System.IO;

namespace builder
{
	public class HtmlFormBuilder : FormBuilder
	{
		private TextWriter writer;
		private int n = 0;

		public HtmlFormBuilder(string outputFileName)
		{
			writer = new StreamWriter(outputFileName);
		}

		public Object StartBuilding() {
			writer.WriteLine("<html>\n<body>\n<form>");
			return null;
		}

		public void EndBuilding() {
			writer.WriteLine("</form>\n</body>\n</html>");
			writer.Close();
		}

		public Object StartGroup(Object parent, string label) {
			writer.WriteLine(label);
			writer.WriteLine("<div style=\"border: solid 1px; float: left\">");
			return null;
		}

		public void EndGroup(Object parent) {
			writer.WriteLine("</div>\n<div style=\"clear: left\" />");
		}

		public void BuildCheckbox(Object parent, string label) {
			writer.WriteLine("<input type=\"checkbox\" />{0}<br />", label);
		}

		public Object StartRadioGroup(Object parent) {
			return "rg" + n++;
		}

		public void BuildRadioButton(Object radioGroup, string label) {
			
			writer.WriteLine("<input type=\"radio\" name=\"{0}\">{1}<br />", radioGroup, label);
		}

		public void EndRadioGroup(Object parent) {
			// nothing to do here
		}

		public void BuildText(Object parent, string label) {
			writer.WriteLine("{0}: <input type=\"text\" /><br />", label);
		}
	}
}

File: various/oom-lab1/HtmlFormBuilder.java
package lab1;

import java.io.BufferedOutputStream;
import java.io.FileOutputStream;
import java.io.PrintWriter;

public class HtmlFormBuilder implements FormBuilder {

	private PrintWriter writer;
	private int n = 0;

	public HtmlFormBuilder(String outputFileName) {
		try {
			writer = new PrintWriter(
					new BufferedOutputStream(
					new FileOutputStream(outputFileName)));
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	public Object startBuilding() {
		writer.write("<html>\n<body>\n<form>\n");
		return null;
	}

	public void endBuilding() {
		writer.write("</form>\n</body>\n</html>\n");
		writer.close();
	}

	public Object startGroup(Object parent, String label) {
		writer.write(label);
		writer.write("\n<div style=\"border: solid 1px; float: left\">\n");
		return null;
	}

	public void endGroup(Object parent) {
		writer.write("</div>\n<div style=\"clear: left\" />\n");
	}

	public Object startRadioGroup(Object parent) {
		return "rg" + n++;
	}

	public void endRadioGroup(Object parent) {
		// nothing to do here
	}

	public void buildCheckbox(Object parent, String label) {
		writer.write("<input type=\"checkbox\" />" + label + "<br />\n");
	}

	public void buildRadioButton(Object radioGroup, String label) {
		writer.write("<input type=\"radio\" name=\"" + radioGroup + "\">" + label + "<br />\n");
	}

	public void buildText(Object parent, String label) {
		writer.write(label + ": <input type=\"text\" /><br />\n");
	}
}
